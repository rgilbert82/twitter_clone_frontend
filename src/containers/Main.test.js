import React from 'react';
import { Main } from '.';
import { shallow } from 'enzyme';

describe(Main, () => {
  let component;

  beforeEach(() => {
    component = shallow(<Main />);
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
