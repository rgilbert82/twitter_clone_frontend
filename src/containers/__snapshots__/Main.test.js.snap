// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Main extends _react2.default.Component {
  render() {
    return (
      _react2.default.createElement(_reactBootstrap.Col, { id: 'main', xs: 12, md: 7, __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this },
        _react2.default.createElement(_reactRouterDom.Switch, { __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this },
          _react2.default.createElement(_reactRouterDom.Route, { exact: true, path: '/', render: () => _react2.default.createElement(_Users.AllUsers, { displayMessage: this.props.displayMessage, __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { exact: true, path: '/register', render: () => _react2.default.createElement(_CurrentUser.Register, { loggedIn: this.props.loggedIn, createUser: this.props.createUser, __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { exact: true, path: '/account', render: () => _react2.default.createElement(_CurrentUser.Account, { loggedIn: this.props.loggedIn, currentUser: this.props.currentUser, editUser: this.props.editUser, deleteAccount: this.props.deleteAccount, __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { path: '/search/', render: () => _react2.default.createElement(_Search.SearchMain, { loggedIn: this.props.loggedIn, currentUser: this.props.currentUser, displayMessage: this.props.displayMessage, navigateToSearch: this.props.navigateToSearch, __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { path: '/users/:slug', render: props => _react2.default.createElement(_Users.UserProfile, Object.assign({}, props, {
              loggedIn: this.props.loggedIn,
              currentUser: this.props.currentUser,
              currentUserChange: this.props.currentUserChange,
              displayMessage: this.props.displayMessage,
              navigateToSearch: this.props.navigateToSearch,
              lowerSidebarTweetCount: this.props.lowerTweetCount,
              raiseSidebarTweetCount: this.props.raiseTweetCount,
              raiseSidebarFollowingCount: this.props.raiseSidebarFollowingCount,
              lowerSidebarFollowingCount: this.props.lowerSidebarFollowingCount, __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this })), __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }))));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "col",
      "children": <Switch>
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/register"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/account"
                render={[Function]}
        />
        <Route
                path="/search/"
                render={[Function]}
        />
        <Route
                path="/users/:slug"
                render={[Function]}
        />
</Switch>,
      "componentClass": "div",
      "id": "main",
      "md": 7,
      "xs": 12,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/register"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/account"
            render={[Function]}
/>,
          <Route
            path="/search/"
            render={[Function]}
/>,
          <Route
            path="/users/:slug"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/register",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/account",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/search/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/users/:slug",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "col",
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/register"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/account"
                    render={[Function]}
          />
          <Route
                    path="/search/"
                    render={[Function]}
          />
          <Route
                    path="/users/:slug"
                    render={[Function]}
          />
</Switch>,
        "componentClass": "div",
        "id": "main",
        "md": 7,
        "xs": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/register"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/account"
              render={[Function]}
/>,
            <Route
              path="/search/"
              render={[Function]}
/>,
            <Route
              path="/users/:slug"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/register",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/account",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/search/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/users/:slug",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
